lib/translator/common.h

```diff
@@ -123,5 +123,22 @@ bool WriteValue(const T& data, Span<uint8_t> out) {
   return true;
 }
 
+// Returns a pointer of type T pointing to buf.data()
+// Uses placement new operator to garentee memory safety
+// Validates alignment and size of buffer
+template <typename T>
+T* SafePointerCast(Span<uint8_t> buf) {
+  if (buf.size() < sizeof(T)) {
+    DebugLog("Pointer cast called on span of invalid size");
+    return nullptr;
+  }
+  if (reinterpret_cast<uintptr_t>(buf.data()) % std::alignment_of<T>::value !=
+      0) {
+    DebugLog("Pointer cast called on unaligned memory");
+    return nullptr;
+  }
+  return new (buf.data()) T;
+}
+
```

Comment:
This is suitable for writing, but not for reading existing data: the contents are default-constructed, which for most of our purposes means undefined. We also need a version that does `Span<const uint8_t>` -> `const T*` which does not use placement new.

Reply:
Rather than using placement new for `const T*`, what method should we use to ensure the memory is not default-constructed?

Reply:
A plain `reinterpret_cast` is appropriate on the reading side. The principle is that we require the caller to have made the necessary arrangements for that to be sound, so we can just cast.